From ead9f65a6a79177ed81da566e5e7bab8da976d14 Mon Sep 17 00:00:00 2001
From: Kenneth Weiss <weiss27@llnl.gov>
Date: Tue, 30 Oct 2018 14:34:46 -0700
Subject: [PATCH] Patch for gtest: Adds guard around definition of
 GTEST_HAS_CXXABI_H_

Don't modify value if it is already defined.

This change is already in gtest's master branch.
See: https://github.com/google/googletest/blob/cf9d6344d28020b70c634c898fe798c862148668/googletest/include/gtest/internal/gtest-port.h#L1005-L1011
---
 .../googletest/include/gtest/internal/gtest-port.h             | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/thirdparty_builtin/googletest-master-2018-02-01/googletest/include/gtest/internal/gtest-port.h b/thirdparty_builtin/googletest-master-2018-02-01/googletest/include/gtest/internal/gtest-port.h
index 1c4e1ba..f250efa 100755
--- a/thirdparty_builtin/googletest-master-2018-02-01/googletest/include/gtest/internal/gtest-port.h
+++ b/thirdparty_builtin/googletest-master-2018-02-01/googletest/include/gtest/internal/gtest-port.h
@@ -982,10 +982,12 @@ using ::std::tuple_size;
 #endif
 
 // _LIBCPP_VERSION is defined by the libc++ library from the LLVM project.
-#if defined(__GLIBCXX__) || (defined(_LIBCPP_VERSION) && !defined(_MSC_VER))
-# define GTEST_HAS_CXXABI_H_ 1
-#else
-# define GTEST_HAS_CXXABI_H_ 0
+#if !defined(GTEST_HAS_CXXABI_H_)
+# if defined(__GLIBCXX__) || (defined(_LIBCPP_VERSION) && !defined(_MSC_VER))
+#  define GTEST_HAS_CXXABI_H_ 1
+# else
+#  define GTEST_HAS_CXXABI_H_ 0
+# endif
 #endif
 
 // A function level attribute to disable checking for use of uninitialized
-- 
2.8.3

